src/dx/pubnub_client.rs

```diff
@@ -172,18 +174,21 @@ where
     /// [`PubNubClient`]: struct.PubNubClient.html
     pub fn build(self) -> Result<PubNubClient<PubNubMiddleware<T>>, PubNubError> {
         self.build_internal()
-            .map(|pre_build| PubNubClient {
-                transport: PubNubMiddleware {
-                    transport: pre_build.transport,
-                    // TODO: String -> Cow<'static, str>
-                    instance_id: pre_build.instance_id.clone(),
-                    user_id: pre_build.config.user_id.clone(),
-                },
-                instance_id: pre_build.instance_id,
-                next_seqn: pre_build.next_seqn,
-                config: pre_build.config,
+            .map_err(|err| ClientInitializationError(err.to_string()))
+            .and_then(|pre_build| {
+                Ok(PubNubClient {
+                    transport: PubNubMiddleware {
+                        transport: pre_build.transport,
+                        // TODO: String -> Cow<'static, str>
+                        instance_id: pre_build.instance_id.clone(),
+                        user_id: pre_build.config.user_id.clone(),
+                        signature_keys: pre_build.config.clone().signature_key_set()?,
+                    },
+                    instance_id: pre_build.instance_id,
+                    next_seqn: pre_build.next_seqn,
+                    config: pre_build.config,
+                })
```

Comment:
Some many clones...

@parfeon I think that `Cow` will be needed quicker than I expected... ;/

[no action needed]

Reply:
Yes. I managed to avoid cloning the body at least, and thought that cloning when creating an instance is acceptable